// @aave/math-utils/dist/cjs/formatters/user/generate-user-reserve-summary.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateUserReserveSummary = void 0;
const pool_math_1 = require("../../pool-math");
function generateUserReserveSummary({ userReserve, marketReferencePriceInUsdNormalized, marketReferenceCurrencyDecimals, currentTimestamp, }) {
    const poolReserve = userReserve.reserve;
    const { priceInMarketReferenceCurrency, decimals } = poolReserve;
    const underlyingBalance = (0, pool_math_1.getLinearBalance)({
        balance: userReserve.scaledATokenBalance,
        index: poolReserve.liquidityIndex,
        rate: poolReserve.liquidityRate,
        lastUpdateTimestamp: poolReserve.lastUpdateTimestamp,
        currentTimestamp,
    });
    const { marketReferenceCurrencyBalance: underlyingBalanceMarketReferenceCurrency, usdBalance: underlyingBalanceUSD, } = (0, pool_math_1.getMarketReferenceCurrencyAndUsdBalance)({
        balance: underlyingBalance,
        priceInMarketReferenceCurrency,
        marketReferenceCurrencyDecimals,
        decimals,
        marketReferencePriceInUsdNormalized,
    });
    const variableBorrows = (0, pool_math_1.getCompoundedBalance)({
        principalBalance: userReserve.scaledVariableDebt,
        reserveIndex: poolReserve.variableBorrowIndex,
        reserveRate: poolReserve.variableBorrowRate,
        lastUpdateTimestamp: poolReserve.lastUpdateTimestamp,
        currentTimestamp,
    });
    const { marketReferenceCurrencyBalance: variableBorrowsMarketReferenceCurrency, usdBalance: variableBorrowsUSD, } = (0, pool_math_1.getMarketReferenceCurrencyAndUsdBalance)({
        balance: variableBorrows,
        priceInMarketReferenceCurrency,
        marketReferenceCurrencyDecimals,
        decimals,
        marketReferencePriceInUsdNormalized,
    });
    return {
        userReserve,
        underlyingBalance,
        underlyingBalanceMarketReferenceCurrency,
        underlyingBalanceUSD,
        variableBorrows,
        variableBorrowsMarketReferenceCurrency,
        variableBorrowsUSD,
        totalBorrows: variableBorrows,
        totalBorrowsMarketReferenceCurrency: variableBorrowsMarketReferenceCurrency,
        totalBorrowsUSD: variableBorrowsUSD,
    };
}
exports.generateUserReserveSummary = generateUserReserveSummary;
