// @bgd-labs/aave-address-book/dist/GhoEthereum.mjs
// src/ts/GhoEthereum.ts
var GHO_TOKEN = "0x40D16FC0246aD3160Ccc09B8D0D3A2cD28aE6C2f";
var GHO_CCIP_TOKEN_POOL = "0x5756880B6a1EAba0175227bf02a7E87c1e02B28C";
var GHO_FLASHMINTER_FACILITATOR = "0xb639D208Bcf0589D54FaC24E655C79EC529762B8";
var GHO_LIQUIDITY_COMMITTEE = "0xA1c93D2687f7014Aaf588c764E3Ce80aF016229b";
var GSM_REGISTRY = "0x167527DB01325408696326e3580cd8e55D99Dc1A";
var GSM_USDC = "0x0d8eFfC11dF3F229AA1EA0509BC9DFa632A13578";
var GSM_USDT = "0x686F8D21520f4ecEc7ba577be08354F4d1EB8262";
var GSM_USDC_FEE_STRATEGY = "0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5";
var GSM_USDT_FEE_STRATEGY = "0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5";
var GSM_USDC_PRICE_STRATEGY = "0x430BEdcA5DfA6f94d1205Cb33AB4f008D0d9942a";
var GSM_USDT_PRICE_STRATEGY = "0x4c707764cbFB4FFa078e169e6b8A6AdbE7526a2c";
var GSM_USDC_ORACLE_SWAP_FREEZER = "0xef6beCa8D9543eC007bceA835aF768B58F730C1f";
var GSM_USDT_ORACLE_SWAP_FREEZER = "0x71381e6718b37C12155CB961Ca3D374A8BfFa0e5";
var CHAIN_ID = 1;
export {
  CHAIN_ID,
  GHO_CCIP_TOKEN_POOL,
  GHO_FLASHMINTER_FACILITATOR,
  GHO_LIQUIDITY_COMMITTEE,
  GHO_TOKEN,
  GSM_REGISTRY,
  GSM_USDC,
  GSM_USDC_FEE_STRATEGY,
  GSM_USDC_ORACLE_SWAP_FREEZER,
  GSM_USDC_PRICE_STRATEGY,
  GSM_USDT,
  GSM_USDT_FEE_STRATEGY,
  GSM_USDT_ORACLE_SWAP_FREEZER,
  GSM_USDT_PRICE_STRATEGY
};
